### PR TITLE
Add creator mod ID for modded enchanted books, potions, arrows, and spawn eggs

### PR DESCRIPTION
**Pull request by mezz**
This lets mods see who created an enchantment, so it can be displayed on tooltips.
Without this it's not always clear that an enchantment is modded, or where it came from.

In the uncommon cases where there are no enchantments or more than one, it just returns the default "Minecraft".

### CODE REVIEW

**Comment by tyler489** [comment]
Yay thanks mezz... lets hope this passes

**Comment by covers1624** [comment]
Is there anything else that needs this? Potions, potion arrows?

**Comment by tyler489** [comment]
Potions and potion arrows SHOULD have the mods in nbt.

**Comment by williewillus** [comment]
Potions and potion arrows have the `PotionType` stored in NBT, which has a registry name, so it can likewise use this improvement.

**Comment by jabelar** [comment]
is mod ID really what we want to display or the mod name? 

EDIT: Never mind. Displaying it was just an example, you're just giving the method to get the id. Looks good.

**Comment by mezz** [comment]
Ah, I knew I was overlooking something.

**Comment by bs2609** [comment]
Spawn eggs?

**Comment by mezz** [comment]
Since there are a few of these to do, I'll see if I can add a forge hook for this instead of making a bunch of patches.

**Comment by mezz** [comment]
Updated with a better patch and support for potions, arrows, and spawn eggs.

File: src/main/java/net/minecraftforge/common/ForgeHooks.java [hunk 1]
```diff
@@ -1386,4 +1396,53 @@ public static void onAdvancement(EntityPlayerMP player, Advancement advancement)
     {
         MinecraftForge.EVENT_BUS.post(new AdvancementEvent(player, advancement));
     }
+
+    /**
+     * Used as the default implementation of {@link Item#getCreatorModId}. Call that method instead.
+     */
+    @Nullable
+    public static String getDefaultCreatorModId(ItemStack itemStack)
```
Comment (bs2609) [inline src/main/java/net/minecraftforge/common/ForgeHooks.java]: Add `@Nonnull` to the `ItemStack` param here?

**Comment by Darkhax** [comment]
I will have to remove this duplicate feature from some of my mods, but I would still be happy to see this PR merged.

File: src/main/java/net/minecraftforge/common/ForgeHooks.java (hunk 1 above)
Comment (mezz) [reply src/main/java/net/minecraftforge/common/ForgeHooks.java]: Good idea

**Pull request merged**